### PR TITLE
Add autosign_source, which may be used instead of autosign_content

### PR DESCRIPTION
**Pull request by downsj2**
This replaces PR #499, it has been rebased to master and squashed.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -89,6 +89,10 @@
 #                                           file('another_module/autosign.sh') or
 #                                           template('another_module/autosign.sh.erb')
 #
+# $autosign_source::                        If set, use this as the source for the autosign file,
+#                                           instead of autosign_content.
+#                                           type:Optional[String]
```
Comment (ekohl) [inline manifests/init.pp]: Now that we use native Puppet 4 types, this line is no longer needed.

File: manifests/server.pp [hunk 2]
```diff
@@ -24,6 +24,10 @@
 #                              file('another_module/autosign.sh') or
 #                              template('another_module/autosign.sh.erb')
 #
+# $autosign_source::           If set, use this as the source for the autosign file,
+#                              instead of autosign_content.
+#                              type:Optional[String]
```
Comment (ekohl) [inline manifests/server.pp]: Same here

Comment (downsj2) [inline manifests/server.pp]: Removed, branch updated.

File: manifests/init.pp [hunk 3]
```diff
@@ -89,6 +89,9 @@
 #                                           file('another_module/autosign.sh') or
 #                                           template('another_module/autosign.sh.erb')
 #
+# $autosign_source::                        If set, use this as the source for the autosign file,
+#                                           instead of autosign_content.
```
Comment (downsj2) [inline manifests/init.pp]: If you specify both content and source, puppet will just reject it.

For example, `puppet apply -e 'file { "/tmp/foo": content => "foo bar\n", source => "/tmp/motd" }'` produces:

`Error: Validation of File[/tmp/foo] failed: You cannot specify more than one of content, source, target at line 1`

While there is certainly value in having a slightly more specific error message, this is something meant for people running the module outside of the Foreman Installer.  I'm not sure it's worth this much hand-holding.  People running the module stand alone should understand basic puppet resources.

This suggestion just seems like over kill to me.

Comment (mmoll) [inline manifests/init.pp]: > If you specify both content and source, puppet will just reject it.

👍 then

**Comment by mmoll** [comment]
merged, thanks @downsj2!

**Pull request merged**